### ISSUE TITLE
Types in https://keystonejs.com/docs/guides/auth-and-access-control are broken

### ISSUE THREAD

**Issue by btakita** [keystonejs/keystone#8940]
Copy & paste the [access control example code](https://keystonejs.com/docs/guides/auth-and-access-control) into `keystone.ts`.

Typescript errors occur.

The types should be designed to not have type errors in the example.

The types should also accommodate named functions. `BaseAccessArgs` is not exported for example.

```ts
function isAdmin({ session }:BaseAccessArgs<BaseListTypeInfo>&{ item: PersonData }) {
  return session?.data.isAdmin
}
```

**Comment by dagrinchi** [comment]
Hello,

I think you are returning a boolean or undefined, You need to force the return as a boolean

```
function isAdmin({ session }: { session?: Session }) {
  return Boolean(session?.data.isAdmin)
}
```

Regards!

**Comment by dagrinchi** [comment]
Waiting for approval to correct the docs:

https://github.com/keystonejs/keystone/pull/9039
https://github.com/keystonejs/keystone/pull/9038

**Comment by dcousens** [comment]
Resolved in #9039, thanks @dagrinchi! :yellow_heart:
